tools/hive-bigquery/bigquery_component.py

```diff
@@ -0,0 +1,725 @@
+                table details
+            gcs_bucket_name (str): GCS bucket name
+        """
+
+        # Uodate this value to increase the maximum number of load job retries
+        bq_load_job_max_retries = 3
+        print_and_log(
+            "Fetching information about BigQuery load jobs from tracking "
+            "table...")
+        query = "SELECT gcs_file_path,bq_job_id,bq_job_retries FROM %s WHERE " \
+                "bq_job_status='RUNNING'" % (
+                    hive_table_model.tracking_table_name)
+        results = mysql_component.execute_query(query)
+        if not results:
+            print_and_log(
+                "No BigQuery job is in RUNNING state. No values to update")
+
+        # Waits till all the load jobs finish
+        while results:
+            count = 0
+            for row in results:
+                gcs_file_path, bq_job_id, bq_job_retries = row
+                # Gets information about the running job
+                job = self.client.get_job(bq_job_id,
+                                          location=self.get_dataset_location(
+                                              bq_table_model.dataset_id))
+
+                if job.state == 'DONE':
+                    # Job finished successfully
+                    if job.errors is None:
+                        query = "UPDATE %s SET bq_job_status='DONE' WHERE " \
+                                "bq_job_id='%s'" % (
+                                    hive_table_model.tracking_table_name,
+                                    bq_job_id)
+                        mysql_component.execute_transaction(query)
+                        print_and_log(
+                            "Updated BigQuery load job {} status RUNNING --> "
+                            "DONE".format(
+                                bq_job_id))
+                        # Deletes the data file in GCS
+                        gcs_component.delete_file(gcs_bucket_name,
+                                                  gcs_file_path)
+                    # Job finished with error
+                    elif job.errors is not None:
+
+                        if bq_job_retries == bq_load_job_max_retries:
+                            query = "UPDATE %s SET bq_job_status='FAILED' " \
+                                    "WHERE bq_job_id='%s'" % (
+                                        hive_table_model.tracking_table_name,
+                                        bq_job_id)
+                            mysql_component.execute_transaction(query)
+                            print_and_log(
+                                "BigQuery job {} failed.Tried for a maximum "
+                                "of 3 times.Updated status RUNNING --> "
+                                "FAILED".format(
+                                    bq_job_id))
+                        else:
+                            query = "UPDATE %s SET bq_job_status='TODO'," \
```

Comment:
Inconsistent use of String formatting. Let's switch this to new format if we can